bundles/org.openhab.binding.bigassfan/src/main/java/org/openhab/binding/bigassfan/internal/discovery/DiscoveryListener.java

```diff
@@ -54,52 +60,68 @@ public DiscoveryListener() throws IOException, SocketException {
         device = new BigAssFanDevice();
         try {
             // Create a socket on the UDP port and get send & receive buffers
-            dSocket = new DatagramSocket(BAF_PORT);
-            dSocket.setSoTimeout(SOCKET_RECEIVE_TIMEOUT);
-            dSocket.setBroadcast(true);
+            DatagramSocket localDatagramSocket = new DatagramSocket(BAF_PORT);
+            localDatagramSocket.setSoTimeout(SOCKET_RECEIVE_TIMEOUT);
+            localDatagramSocket.setBroadcast(true);
+            dSocket = localDatagramSocket;
             rcvBuffer = new byte[256];
             rcvPacket = new DatagramPacket(rcvBuffer, rcvBuffer.length);
             bcastAddress = InetAddress.getByName(BCAST_ADDRESS);
             bcastBuffer = POLL_MESSAGE.getBytes(StandardCharsets.US_ASCII);
             bcastPacket = new DatagramPacket(bcastBuffer, bcastBuffer.length, bcastAddress, BAF_PORT);
-        } catch (UnknownHostException uhe) {
-            logger.warn("UnknownHostException sending poll request for fans: {}", uhe.getMessage(), uhe);
+        } catch (UnknownHostException | SocketException | SecurityException e) {
+            logger.warn("Unexpected exception sending poll request for fans: {}", e.getMessage(), e);
         }
     }
 
     public BigAssFanDevice waitForMessage() throws IOException, SocketTimeoutException {
         // Wait to receive a packet
-        rcvPacket.setLength(rcvBuffer.length);
-        dSocket.receive(rcvPacket);
+        DatagramPacket localPacket = rcvPacket;
+        DatagramSocket localDatagramSocket = dSocket;
+
+        if (localPacket != null) {
+            localPacket.setLength(rcvBuffer.length);
+        }
+
+        if (localDatagramSocket != null && localPacket != null) {
+            localDatagramSocket.receive(rcvPacket);
```

Comment:
I assume you meant:
```suggestion
            localDatagramSocket.receive(localPacket);
```
?

Reply:
good catch! thanks